commit 4558c8f84b2f8d3ba1483727bcb49935ae8ff595
Author: Jeff King <peff@peff.net>
Date:   Wed Jun 4 16:56:22 2025 -0400

    curl: fix symbolic constant typechecks with curl_easy_setopt()
    
    As with the previous two commits, we should be passing long integers,
    not regular ones, to curl_easy_setopt(), and compiling against curl 8.14
    loudly complains if we don't.
    
    This patch catches the remaining cases, which are ones where we pass
    curl's own symbolic constants. We'll cast them to long manually in each
    call.
    
    It seems kind of weird to me that curl doesn't define these constants as
    longs, since the point of them is to pass to curl_easy_setopt(). But in
    the curl documentation and examples, they clearly show casting them as
    part of the setopt calls. It may be that there is some reason not to
    push the type into the macro, like backwards compatibility. I didn't
    dig, as it doesn't really matter: we have to follow what existing curl
    versions ask for anyway.
    
    Signed-off-by: Jeff King <peff@peff.net>
    Signed-off-by: Junio C Hamano <gitster@pobox.com>

diff --git a/git-core/http.c b/git-core/http.c
index cce2ea7287..ecbc47ea4b 100644
--- a/git-core/http.c
+++ b/git-core/http.c
@@ -1118,7 +1118,7 @@ static CURL *get_curl_handle(void)
 	}
 
 	curl_easy_setopt(result, CURLOPT_MAXREDIRS, 20L);
-	curl_easy_setopt(result, CURLOPT_POSTREDIR, CURL_REDIR_POST_ALL);
+	curl_easy_setopt(result, CURLOPT_POSTREDIR, (long)CURL_REDIR_POST_ALL);
 
 #ifdef GIT_CURL_HAVE_CURLOPT_PROTOCOLS_STR
 	{
@@ -1193,18 +1193,18 @@ static CURL *get_curl_handle(void)
 
 		if (starts_with(curl_http_proxy, "socks5h"))
 			curl_easy_setopt(result,
-				CURLOPT_PROXYTYPE, CURLPROXY_SOCKS5_HOSTNAME);
+				CURLOPT_PROXYTYPE, (long)CURLPROXY_SOCKS5_HOSTNAME);
 		else if (starts_with(curl_http_proxy, "socks5"))
 			curl_easy_setopt(result,
-				CURLOPT_PROXYTYPE, CURLPROXY_SOCKS5);
+				CURLOPT_PROXYTYPE, (long)CURLPROXY_SOCKS5);
 		else if (starts_with(curl_http_proxy, "socks4a"))
 			curl_easy_setopt(result,
-				CURLOPT_PROXYTYPE, CURLPROXY_SOCKS4A);
+				CURLOPT_PROXYTYPE, (long)CURLPROXY_SOCKS4A);
 		else if (starts_with(curl_http_proxy, "socks"))
 			curl_easy_setopt(result,
-				CURLOPT_PROXYTYPE, CURLPROXY_SOCKS4);
+				CURLOPT_PROXYTYPE, (long)CURLPROXY_SOCKS4);
 		else if (starts_with(curl_http_proxy, "https")) {
-			curl_easy_setopt(result, CURLOPT_PROXYTYPE, CURLPROXY_HTTPS);
+			curl_easy_setopt(result, CURLOPT_PROXYTYPE, (long)CURLPROXY_HTTPS);
 
 			if (http_proxy_ssl_cert)
 				curl_easy_setopt(result, CURLOPT_PROXY_SSLCERT, http_proxy_ssl_cert);
